lib/SPIRV/libSPIRV/SPIRVEntry.cpp

```diff
@@ -358,7 +371,21 @@ SPIRVEntry::getMemberDecorationStringLiteral(Decoration Kind,
   for (SPIRVWord I = 0; I < Loc->second->getLiteralCount(); ++I)
     Literals.push_back(Loc->second->getLiteral(I));
 
-  return getString(Literals.cbegin(), Literals.cend());
+  return getString(Literals);
+}
+
+std::vector<std::string>
+SPIRVEntry::getMemberDecorationStringLiterals(Decoration Kind,
+                                              SPIRVWord MemberNumber) const {
+  std::vector<SPIRVWord> Literals;
+  auto Loc = MemberDecorates.find({MemberNumber, Kind});
+  if (Loc == MemberDecorates.end())
+    return {};
+
+  for (SPIRVWord I = 0; I < Loc->second->getLiteralCount(); ++I)
```

Comment:
I think we can avoid copying of a vector if we move this loop to `SPIRVDecorateGeneric` class.